CMakeLists.txt

```diff
@@ -2560,84 +2523,108 @@ find_package(Qt${QT_VERSION_MAJOR}
-  Qt${QT_VERSION_MAJOR}::Concurrent
-  Qt${QT_VERSION_MAJOR}::Core
-  Qt${QT_VERSION_MAJOR}::Gui
-  Qt${QT_VERSION_MAJOR}::Network
-  Qt${QT_VERSION_MAJOR}::OpenGL
-  Qt${QT_VERSION_MAJOR}::PrintSupport
-  Qt${QT_VERSION_MAJOR}::Qml
-  Qt${QT_VERSION_MAJOR}::QuickWidgets
-  Qt${QT_VERSION_MAJOR}::Sql
-  Qt${QT_VERSION_MAJOR}::Svg
-  Qt${QT_VERSION_MAJOR}::Test
-  Qt${QT_VERSION_MAJOR}::Widgets
-  Qt${QT_VERSION_MAJOR}::Xml)
+foreach(COMPONENT ${QT_COMPONENTS})
+  target_link_libraries(mixxx-lib PUBLIC Qt6::${COMPONENT})
+endforeach()
 if(QT_EXTRA_COMPONENTS)
   foreach(COMPONENT ${QT_EXTRA_COMPONENTS})
     target_link_libraries(mixxx-lib PUBLIC Qt6::${COMPONENT})
   endforeach()
 endif()
 
-if(QML)
-  set(QT_QML_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/qml)
-  set_target_properties(mixxx-lib PROPERTIES AUTOMOC ON)
-  qt_add_qml_module(mixxx-lib
-    URI Mixxx
-    VERSION 1.0
-    RESOURCE_PREFIX /mixxx.org/imports
-    IMPORTS QtQuick
-    QML_FILES
-      res/qml/Mixxx/MathUtils.mjs
-      res/qml/Mixxx/PlayerDropArea.qml
-    NO_GENERATE_QMLDIR
-  )
-
-  # Generation of the `qmldir` file has been disabled (via `NO_GENERATE_QMLDIR`
-  # above) due to QTBUG-100326, which breaks JavaScript module imports:
-  # https://bugreports.qt.io/browse/QTBUG-100326
-  # Instead, a handwritten `qmldir` file that works around the issue needs to
-  # be added to the resources here.
-  set_source_files_properties("res/qml/Mixxx/qmldir" PROPERTIES QT_RESOURCE_ALIAS "qmldir")
-  qt_add_resources(mixxx-libplugin qmldir
-    FILES "res/qml/Mixxx/qmldir"
-    PREFIX "/mixxx.org/imports/Mixxx"
-  )
-  # to make also qmllint happy
-  configure_file(res/qml/Mixxx/qmldir qml/Mixxx/qmldir COPYONLY)
-
-  # FIXME: Currently we need to add these include directories due to
-  # QTBUG-87221. We should figure out a better way to fix this.
-  # See: https://bugreports.qt.io/browse/QTBUG-87221
-  target_include_directories(mixxx-lib PRIVATE src/control src/qml)
-  target_link_libraries(mixxx-lib PRIVATE mixxx-libplugin)
-
-  qt_add_library(mixxx-qml-mixxxcontrols STATIC)
-  set_target_properties(mixxx-qml-mixxxcontrols PROPERTIES AUTOMOC ON)
-  qt_add_qml_module(mixxx-qml-mixxxcontrols
-    URI Mixxx.Controls
-    VERSION 1.0
-    RESOURCE_PREFIX /mixxx.org/imports
-    OPTIONAL_IMPORTS Mixxx
-    QML_FILES
-      res/qml/Mixxx/Controls/Knob.qml
-      res/qml/Mixxx/Controls/Slider.qml
-      res/qml/Mixxx/Controls/Spinny.qml
-      res/qml/Mixxx/Controls/WaveformOverviewHotcueMarker.qml
-      res/qml/Mixxx/Controls/WaveformOverviewMarker.qml
-      res/qml/Mixxx/Controls/WaveformOverview.qml
-  )
-  target_link_libraries(mixxx-lib PRIVATE mixxx-qml-mixxxcontrolsplugin)
-
-  # qt_finalize_target takes care that the resources :/mixxx.org/imports/Mixxx/
-  # and :/mixxx.org/imports/Mixxx/Controls are placed into beginning of the binary
-  qt_finalize_target(mixxx)
+set(QT_QML_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/qml)
+qt_add_library(mixxx-qml-lib STATIC)
```

Comment:
Do we need an extra library for widgets only?

Reply:
It looks like this is the only option yes. My very limited knowledge and understanding of the problem seems to say that as soon as you configure your target with `qt_add_qml_module`, some of the autogen process will not work with legacy UI. My gut feeling is that `uic` or `moc` gets somehow disabled or altered and thus break the single library with both widget type.
I believe there may be more elegant way to achieve that, but sadly, I mentioned on Zulip, I'm a newbie when it comes to `CMake`, especially `CMake` + `Qt`, so if anyone can suggest a different simple approach, I'm happy to try it.